interviewChallenge.js

```diff
@@ -0,0 +1,55 @@
+// const flatten = (elements) => {
+//   // Flattens a nested array (the nesting can be to any depth).
+//   // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
+//   const flat = [].concat(elements)
+//   flat.(Array.isArray) ? flatten(flat) : flat;
+//  elements.forEach((element) => {
+//      if (Array.isArray(element) {
+//          smush it
+//          add it to a new array 'container'
+//      })
+//      return the container
+//  })
+
+// };
+
+// flatten([1, [2], [3, [[4]]]]);
+const flatten = (elements) => {
```

Comment:
The idea here is sound. It can definitely be cleaned up though. You can check out the solution implementation to see some slick implementations.